app/controllers/api/base_controller/renderer.rb

```diff
@@ -350,15 +352,19 @@ def attr_accessible?(object, attr)
       def attr_virtual?(object, attr)
         return false if ID_ATTRS.include?(attr)
         primary = attr_split(attr).first
-        (object.class.respond_to?(:reflect_on_association) && object.class.reflect_on_association(primary)) ||
-          (object.class.respond_to?(:virtual_attribute?) && object.class.virtual_attribute?(primary)) ||
-          (object.class.respond_to?(:virtual_reflection?) && object.class.virtual_reflection?(primary))
+        klass   = object
+        klass   = object.class if object.kind_of?(ActiveRecord::Base)
+        (klass.respond_to?(:reflect_on_association) && klass.reflect_on_association(primary)) ||
```

Comment:
`klass.try(:reflect_on_association)` ?  The .try does a respond_to check

Reply:
Though I do see the old code had it that way.

Reply:
Yeah, I didn't change the code here on purpose.  I was just refactoring it to allow it to accept a class directly instead of an instance and converting it to a class.

Explained already here:  https://github.com/ManageIQ/manageiq-api/pull/887#discussion_r482307292